pkgs/by-name/si/simple64/package.nix

```diff
@@ -0,0 +1,99 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl,
+  writeShellScriptBin,
+  cmake,
+  ninja,
+  pkg-config,
+  makeWrapper,
+  zlib,
+  libpng,
+  SDL2,
+  SDL2_net,
+  hidapi,
+  qt6,
+  vulkan-loader,
+}:
+
+let
+  cheats-json = fetchurl {
+    url = "https://raw.githubusercontent.com/simple64/cheat-parser/87963b7aca06e0d4632b66bc5ffe7d6b34060f4f/cheats.json";
+    hash = "sha256-rS/4Mdi+18C2ywtM5nW2XaJkC1YnKZPc4YdQ3mCfESU=";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "simple64";
+  version = "2024.03.2";
+
+  src = fetchFromGitHub {
+    owner = "simple64";
+    repo = "simple64";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-ECgs1vjEA9qsohtS8J+42YztVb+hua3HLIha3mNz9gs=";
+  };
+
+  patches = [
+    ./dont-use-vosk-and-discord.patch
+    ./add-official-server-error-message.patch
+  ];
+
+  postPatch = ''
+    cp ${cheats-json} cheats.json
+  '';
+
+  nativeBuildInputs = [
+    qt6.wrapQtAppsHook
+    cmake
+    ninja
+    pkg-config
+    makeWrapper
+    # fake git command for version info generator
+    (writeShellScriptBin "git" "echo ${finalAttrs.src.rev}")
+  ];
+
+  buildInputs = [
+    zlib
+    libpng
+    SDL2
+    SDL2_net
+    hidapi
+    qt6.qtbase
+    qt6.qtwebsockets
+  ];
```

Comment:
`qt6.qtwayland` should be added here for Wayland support.

Reply:
Looks like this is not mentioned anywhere upstream.
Is this just so that it gets added to `QT_PLUGIN_PATH` and `NIXPKGS_QT6_QML_IMPORT_PATH`, and that makes it work on wayland?

Reply:
`qtwayland` allow the client to use the Wayland platform plugin, without it it will only run in x11/XWayland mode.

https://nixos.org/manual/nixpkgs/unstable/#sec-language-qt

Reply:
Thanks for the link and the explanation.